### PR TITLE
Feature/browse no login

### PR DESCRIPTION
**Pull request by jamesjarvis**
Comment for issues, I dunno whether 'continue to browse' is maybe the right wording?

### CODE REVIEW

File: src/app/components/Core/Layout/NavigationBar/Logout/Logout.jsx [hunk 1]
```diff
@@ -8,23 +8,34 @@ class Logout extends Component {
...
 Logout.propTypes = {
   mk: PropTypes.any.isRequired,
 };
 
-export default withMK(Logout);
+const bindings = {
+  [MusicKit.Events.authorizationStatusDidChange]: 'authorization',
+};
+
+export default withMK(Logout, bindings);
```
Comment (raphaelvigee) [inline src/app/components/Core/Layout/NavigationBar/Logout/Logout.jsx]: `Logout` had a `login` function ??
=> The class/file should be renamed

File: src/app/components/Core/Layout/NavigationBar/Search/SearchBar.jsx [hunk 2]
```diff
@@ -81,14 +81,20 @@ class SearchBar extends React.Component {
   }
 
   async searchLibrary(query) {
-    const libraryData = await this.props.mk.instance.api.library.search(query, {
-      types: ['library-albums', 'library-songs', 'library-playlists', 'library-artists'],
-      limit: 3,
-    });
+    if (this.props.mk.instance.isAuthorized) {
```
Comment (raphaelvigee) [inline src/app/components/Core/Layout/NavigationBar/Search/SearchBar.jsx]: This should be outside of this function. The function should never be called

File: src/app/components/Core/Layout/Sidebar/Sidebar.jsx [hunk 3]
```diff
@@ -34,27 +41,40 @@ export default class Sidebar extends React.Component {
...
+        title={'My Library'}
+        items={[
+          { to: '/artists', label: 'Artists' },
+          { to: '/albums', label: 'Albums' },
+          { to: '/songs', label: 'Songs' },
+          { to: '/playlists', label: 'Playlists' },
+        ]}
+      />
+    ) : (
+      <></>
```
Comment (raphaelvigee) [inline src/app/components/Core/Layout/Sidebar/Sidebar.jsx]: Remove that

File: src/app/components/Core/Player/Player.jsx [hunk 4]
```diff
@@ -153,7 +153,9 @@ class Player extends React.Component {
       mediaItem: { item: nowPlayingItem },
       playbackTime,
     } = this.props.mk;
-    const duration = nowPlayingItem.playbackDuration / 1000;
+    const duration = this.props.mk.instance.isAuthorized
```
Comment (raphaelvigee) [inline src/app/components/Core/Player/Player.jsx]: Why ?

File: src/app/components/Core/Player/Player.jsx [hunk 5]
```diff
@@ -179,7 +181,7 @@ class Player extends React.Component {
         onMouseDown={this.onStartScrubbing}
         onMouseUp={this.onEndScrubbing}
         min={0}
-        max={nowPlayingItem.playbackDuration}
+        max={duration * 1000}
```
Comment (raphaelvigee) [inline src/app/components/Core/Player/Player.jsx]: Pretty sure this should be multiplied by 1000

File: src/app/components/Login/Login.jsx [hunk 6]
```diff
@@ -1,61 +1,87 @@
...
-      </span>
-      <div className={classes.disclaimer}>
-        <div className={classes.disclaimerIcon}>
-          <span>
-            <i className="fas fa-info icon" />
-          </span>
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.loginNavigate = this.loginNavigate.bind(this);
```
Comment (raphaelvigee) [inline src/app/components/Login/Login.jsx]: \n

File: src/app/components/MusicKitAuthorizeProvider.jsx [hunk 7]
```diff
@@ -10,6 +10,7 @@ class MusicKitAuthorizeProvider extends React.Component {
 
     this.state = {
       ready: false,
+      browsing: false,
```
Comment (raphaelvigee) [inline src/app/components/MusicKitAuthorizeProvider.jsx]: This should use context

Comment (raphaelvigee) [reply src/app/components/MusicKitAuthorizeProvider.jsx]: @BrychanOdlum

File: src/app/components/Core/Layout/NavigationBar/Authorize/Authorize.jsx [hunk 8]
```diff
@@ -20,22 +21,20 @@ class Logout extends Component {
...
 
-Logout.propTypes = {
+Authorize.propTypes = {
   mk: PropTypes.any.isRequired,
 };
 
-const bindings = {
-  [MusicKit.Events.authorizationStatusDidChange]: 'authorization',
-};
+Authorize.contextType = AuthorizeContext;
```
Comment (raphaelvigee) [inline src/app/components/Core/Layout/NavigationBar/Authorize/Authorize.jsx]: Use the `withContext` HOC

File: src/app/components/Core/Layout/NavigationBar/Authorize/Authorize.jsx [hunk 9]
```diff
@@ -20,22 +21,20 @@ class Logout extends Component {
...
 
-export default withMK(Logout, bindings);
```
Comment (raphaelvigee) [inline src/app/components/Core/Layout/NavigationBar/Authorize/Authorize.jsx]: Why removing the bindings ?

File: src/app/components/Core/Layout/NavigationBar/Search/SearchBar.jsx [hunk 10]
```diff
@@ -170,4 +168,6 @@ SearchBar.propTypes = {
   mk: PropTypes.any.isRequired,
 };
 
+SearchBar.contextType = AuthorizeContext;
```
Comment (raphaelvigee) [inline src/app/components/Core/Layout/NavigationBar/Search/SearchBar.jsx]: Same as above

File: src/app/components/MusicKitAuthorizeProvider.jsx [hunk 11]
```diff
@@ -57,14 +58,18 @@ class MusicKitAuthorizeProvider extends React.Component {
...
+        <SplashScreen
           onClick={() => this.props.mk.instance.authorize()}
           onBrowse={() => this.setState({ browsing: true })}
         />
       );
     }
 
-    return this.props.children;
+    return (
+      <AuthorizeContext.Provider value={isAuthorized}>
```
Comment (raphaelvigee) [inline src/app/components/MusicKitAuthorizeProvider.jsx]: The provider should be in App

File: src/app/components/SplashScreen/SplashScreen.scss [hunk 12]
```diff
@@ -1,8 +1,8 @@
 @import '../../styles/settings/colors';
 
-.loginContainer {
-  background: $login_background_color
-    linear-gradient(315deg, #cd3c5a, $login_background_color);
+.splashContainer {
+  background: $splash_background_color
+    linear-gradient(315deg, #cd3c5a, $splash_background_color);
```
Comment (raphaelvigee) [inline src/app/components/SplashScreen/SplashScreen.scss]: Bad indent

File: src/app/components/MusicKitAuthorizeProvider.jsx (hunk 11 above)
Comment (jamesjarvis) [reply src/app/components/MusicKitAuthorizeProvider.jsx]: Putting the provider in the app would mean adding the event listener into the App. As this component was already listening for the authorization did change event, I think its best to keep the provider in this component.

File: src/app/components/SplashScreen/SplashScreen.scss (hunk 12 above)
Comment (jamesjarvis) [reply src/app/components/SplashScreen/SplashScreen.scss]: my linter isnt showing any errors

File: src/app/components/MusicKitAuthorizeProvider.jsx [hunk 13]
```diff
@@ -66,7 +66,7 @@ class MusicKitAuthorizeProvider extends React.Component {
     }
 
     return (
-      <AuthorizeContext.Provider value={isAuthorized}>
+      <AuthorizeContext.Provider value={{ authorized: isAuthorized }}>
```
Comment (raphaelvigee) [inline src/app/components/MusicKitAuthorizeProvider.jsx]: Move the provider to `App`

Comment (jamesjarvis) [reply src/app/components/MusicKitAuthorizeProvider.jsx]: lol no thanks, lets just approve it yeh

**Pull request merged**